src/kvtree_io.c

```diff
@@ -135,7 +135,10 @@ int kvtree_close(const char* file, int fd)
   return KVTREE_SUCCESS;
 }
 
-int kvtree_file_lock_write(const char* file, int fd)
+/*
+ * Provide thread-safe and process-safe locking to a kvtree file
+ */
+int kvtree_file_lock_read_write(const char* file, int fd, int write)
 {
   #ifdef KVTREE_FILE_LOCK_USE_FLOCK
     if (flock(fd, LOCK_EX) != 0) {
```

Comment:
In case the user opts to use flock, we should add ``LOCK_SH``.

Reply:
Good catch!